src/main.rs

```diff
@@ -1,17 +1,30 @@
+use notes_r_us::backend;
 use poem::{self, endpoint::StaticFilesEndpoint, listener::TcpListener, Route, Server};
+use poem_openapi::OpenApiService;
 use std::env::{self};
-// use poem_openapi::{self, OpenApi}
 
 #[tokio::main]
-async fn main() {
-    let app = Route::new().nest(
-        "/",
-        StaticFilesEndpoint::new(env::current_dir().unwrap().join("notes_r_us_ui/dist"))
-            .index_file("index.html"),
-    );
+async fn main() -> std::result::Result<(), std::io::Error> {
+    let api_service = OpenApiService::new(
+        backend::Api,
+        "Notes R Us API Documentation",
+        std::env::var("CARGO_PACKAGE_VERSION").unwrap_or(String::from("N/A")),
+    )
+    .server("http://localhost:80")
```

Comment:
Should we add a Variable that holds the local server end point in one spot?

E.g.
```rust
let sever_str = format!("{}:{}", "0.0.0.0", 8000).as_str()
```

Also please pick a non http port for serving the server.

Reply:
Port 8000 is in use in my system, I will choose 3000 as that's what it was before, also for the server endpoint variable, we could have it be `localhost` if the program is compiled in `debug` mode and make it the actual URL if it's compiled in `release` mode.

What do you reckon?

Reply:
@Notliam99 what is going to be the URL

![image](https://github.com/l2dit/NotesRUs/assets/95382758/4913224f-08ca-4379-9872-72b0ec07c84e)

Reply:
Renamed var to `server_url` , as its only going to contain URL not port and https:// | http://.

Reply:
I think we'll make it notes.nzdev.org I'd have to move my notes from this subdomain tho.

Or something like notesrus.nzdev.org.